src/embeddings/embeddings_calculator.cc

```diff
@@ -124,31 +128,60 @@ class EmbeddingsCalculator : public CalculatorBase {
                     ov::Shape{received_batch_size},
                     strings->data()};
                 tokenizerOutputMap = tokenizer_session->infer(tokenizerInputMap);
-            } else {
-                // TODO: input already tokenized
-                return absl::InvalidArgumentError("not implemented");
-            }
-            ::InferenceInput embeddingsInputMap;
-            // Check if tokenizer produced at least the number of outputs as there are inputs in embedding model
-            RET_CHECK(tokenizerOutputMap.size() >= embeddingsInputNames.size());
-            for (const auto& embeddingsInputName : embeddingsInputNames) {
-                auto it = tokenizerOutputMap.find(embeddingsInputName);
-                RET_CHECK(it != tokenizerOutputMap.end());
-                SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Embedding model input {} is connected with matching tokenizer output", embeddingsInputName);
-                embeddingsInputMap[embeddingsInputName] = it->second;
+                // Check if tokenizer produced at least the number of outputs as there are inputs in embedding model
+                RET_CHECK(tokenizerOutputMap.size() >= embeddingsInputNames.size());
+                for (const auto& embeddingsInputName : embeddingsInputNames) {
+                    auto it = tokenizerOutputMap.find(embeddingsInputName);
+                    RET_CHECK(tokenizerOutputMap[EMBEDDINGS_MODEL_INPUT_IDS_NAME].get_shape().size() == 2);
```

Comment:
This ret check is way too early, at this point you have no idea if EMBEDDINGS_MODEL_INPUT_IDS_NAME exists in tokenizerOutputMap. Therefore using `operator[]` will insert new element to the output map which is not desired

Reply:
What is more, I don't think we need to repeat this RET_CHECK N times, where N is equal to embeddingsInputNames.size() (for loop)